ledger/db/src/lib.rs

```diff
@@ -49,10 +49,27 @@ pub const KEY_IMAGES_BY_BLOCK_DB_NAME: &str = "ledger_db:key_images_by_block";
 pub const TX_OUTS_BY_BLOCK_DB_NAME: &str = "ledger_db:tx_outs_by_block";
 pub const BLOCK_NUMBER_BY_TX_OUT_INDEX: &str = "ledger_db:block_number_by_tx_out_index";
 
-// Keys used by the `counts` database.
+/// Keys used by the `counts` database.
 pub const NUM_BLOCKS_KEY: &str = "num_blocks";
 
-// The value stored for each entry in the `tx_outs_by_block` database.
+/// Metadata store settings that are used for version control.
+#[derive(Clone, Default, Debug)]
+pub struct LedgerDbMetadataStoreSettings;
+impl MetadataStoreSettings for LedgerDbMetadataStoreSettings {
```

Comment:
This stuff is the main idea behind the work proposed in this PR. Each LMDB user would need to have this, instead of duplicating the entire metadata_store.rs file.